packages/server-wallet/src/protocols/open-channel.ts

```diff
@@ -122,7 +122,7 @@ const signPostFundSetup = (ps: ProtocolState): ProtocolResult | false =>
   });
 
 const completeOpenChannel = (ps: ProtocolState): CompleteObjective | false =>
-  (isRunning(ps.app.supported) || isPostfundSetup(ps.app.supported)) &&
+  (isRunning(ps.app.supported) || ps.app.supported === ps.app.participants.length * 2 - 1) &&
```

Comment:
```suggestion
  (isRunning(ps.app.supported) || ps.app.supported.turnNum === ps.app.participants.length * 2 - 1) &&
```

This is a typo right?

Reply:
This should have been a type error. The type of `ps.app.supported` appears to be `undefined` on this line though ...

<img width="333" alt="Screen Shot 2020-10-24 at 12 35 29 PM" src="https://user-images.githubusercontent.com/8432675/97090963-7c35bf00-15f5-11eb-8092-4e2d9c0dfe73.png">

Reply:
Yeah there is a general code cleanup that can be done to these protocols to assert certain properties inside the smaller functions the `protocol` function is composed of that would ensure it is defined.

Reply:
This is really counterintuitive:
<img width="585" alt="Screen Shot 2020-10-24 at 12 42 54 PM" src="https://user-images.githubusercontent.com/8432675/97091099-79879980-15f6-11eb-87b2-dd4c0c7038f0.png">

Reply:
I think it has something to do with non-structural / nominal typing and maybe the lack of negated types

Reply:
typescript is structurally typed, and not nominally typed?

I didn't expect `barGuard(foobar: Foo | Bar): foobar is Bar` to change the type at all if `barGuard` returned `false`.

Reply:
This is perhaps an argument in favour of the `Option(State)` type instead of `State | undefined`?

Reply:
so the [example](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) in the docs is pretty similar to the `isSomething` example.

It seems, then, that the mistake is that `isRunning` is a _type guard_ that asserts that it is a `State` when it returns `true`. The compiler then narrows it down to the remaining allowed types, when it returns false, which is `undefined`.

Instead, `isRunning` should narrow it to a `RunningState` type, leaving open the possibility that it is a `NotRunningState | undefined`.